internal/testutil/pkgdata.go

```diff
@@ -1,152 +1,327 @@
+	compressExt string // "gz", "xz", "zst" or empty, empty means "zst"
+	readSource  func(pkg *pkgDef, path string) ([]byte, error)
+	tarTemplate tar.Header
+	entries     []tarEntry
+}
+
+var defaultTarTemplate = tar.Header{
+	Uid:     0,
+	Gid:     0,
+	Uname:   "root",
+	Gname:   "root",
+	ModTime: time.Unix(0, 0),
+	Format:  tar.FormatGNU,
+}
+
+var pkgs = map[string]pkgDef{
+	"base-files": pkgDef{
+		entries: []tarEntry{
+			{name: "./"},
+			{name: "./bin/"},
+			{name: "./boot/"},
+			{name: "./dev/"},
+			{name: "./etc/"},
+			{name: "./etc/debian_version"},
+			{name: "./etc/default/"},
+			{name: "./etc/dpkg/"},
+			{name: "./etc/dpkg/origins/"},
+			{name: "./etc/dpkg/origins/debian"},
+			{name: "./etc/dpkg/origins/ubuntu"},
+			{name: "./etc/host.conf"},
+			{name: "./etc/issue"},
+			{name: "./etc/issue.net"},
+			{name: "./etc/legal"},
+			{name: "./etc/lsb-release"},
+			{name: "./etc/profile.d/"},
+			{name: "./etc/profile.d/01-locale-fix.sh"},
+			{name: "./etc/skel/"},
+			{name: "./etc/update-motd.d/"},
+			{name: "./etc/update-motd.d/00-header", mode: 00755},
+			{name: "./etc/update-motd.d/10-help-text", mode: 00755},
+			{name: "./etc/update-motd.d/50-motd-news", mode: 00755},
+			{name: "./home/"},
+			{name: "./lib/"},
+			{name: "./lib/systemd/"},
+			{name: "./lib/systemd/system/"},
+			{name: "./lib/systemd/system/motd-news.service"},
+			{name: "./lib/systemd/system/motd-news.timer"},
+			{name: "./proc/"},
+			{name: "./root/", mode: 00700},
+			{name: "./run/"},
+			{name: "./sbin/"},
+			{name: "./sys/"},
+			{name: "./tmp/", mode: 01777},
+			{name: "./usr/"},
+			{name: "./usr/bin/"},
+			{name: "./usr/bin/hello", mode: 00775},
+			{name: "./usr/games/"},
+			{name: "./usr/include/"},
+			{name: "./usr/lib/"},
+			{name: "./usr/lib/os-release"},
+			{name: "./usr/sbin/"},
+			{name: "./usr/share/"},
+			{name: "./usr/share/dict/"},
+			{name: "./usr/share/doc/"},
+			{name: "./usr/share/doc/base-files/"},
+			{name: "./usr/share/doc/base-files/copyright"},
+			{name: "./usr/share/info/"},
+			{name: "./usr/share/man/"},
+			{name: "./usr/share/misc/"},
+			{name: "./usr/src/"},
+			{name: "./var/"},
+			{name: "./var/backups/"},
+			{name: "./var/cache/"},
+			{name: "./var/lib/"},
+			{name: "./var/lib/dpkg/"},
+			{name: "./var/lib/misc/"},
+			{name: "./var/local/", mode: 02775},
+			{name: "./var/lock/", mode: 01777},
+			{name: "./var/log/"},
+			{name: "./var/run/"},
+			{name: "./var/spool/"},
+			{name: "./var/tmp/", mode: 01777},
+			{name: "./etc/os-release", linkname: "../usr/lib/os-release"},
+		},
+	},
+}
+
+//go:embed all:pkgdata
+var pkgdataFS embed.FS
+
+func (pkg *pkgDef) init(pkgName string) error {
+	if pkg.tarTemplate.Format == 0 {
+		pkg.tarTemplate = defaultTarTemplate
+	}
+	if pkg.sourceDir == "" {
+		pkg.sourceDir = "pkgdata/" + pkgName
+	}
+	if pkg.readSource == nil {
+		pkg.readSource = func(pkg *pkgDef, path string) ([]byte, error) {
+			localPath := filepath.Join(pkg.sourceDir, path)
```

Comment:
This abstraction doesn't look necessary. This is a private type used to produce testing data, so we can simply inline the content here and avoid some complexity.

Reply:
Dropped, content is inlined now.